open-metadata-implementation/access-services/subject-area/subject-area-client/src/main/java/org/odpi/openmetadata/accessservices/subjectarea/client/SubjectAreaClient.java

```diff
@@ -9,65 +9,191 @@
     T getByGUID(String userId, String guid) throws InvalidParameterException,
                                                    PropertyServerException,
                                                    UserNotAuthorizedException;
 
+    /**
+     * Create a object. To create, you must pass the created object and specify a unique user identifier.
+     *
+     * @param userId   unique identifier for requesting user, under which the request is performed.
+     * @param supplied object to create.
+     * @return created object.
+     *
+     * @throws PropertyServerException    something went wrong with the REST call stack.
+     * @throws UserNotAuthorizedException the requesting user is not authorized to issue this request.
+     * @throws InvalidParameterException  one of the parameters is null or invalid.
+     */
     T create(String userId, T supplied) throws InvalidParameterException,
                                                PropertyServerException,
                                                UserNotAuthorizedException;
 
+    /**
+     * Request to find all objects of the type T.
+     * Getting all objects when there are 100s of thousands may be will lill the system.
+     *
+     * @param userId unique identifier for requesting user, under which the request is performed.
+     * @return list all objects of the T type.
+     *
+     * @throws PropertyServerException    something went wrong with the REST call stack.
+     * @throws UserNotAuthorizedException the requesting user is not authorized to issue this request.
+     * @throws InvalidParameterException  one of the parameters is null or invalid.
+     */
     default List<T> findAll(String userId) throws InvalidParameterException,
                                                   PropertyServerException,
                                                   UserNotAuthorizedException
     {
        return find(userId, EMPTY_FIND_REQUEST);
     }
 
+    /**
+     * Request to find entities of the type T.
+     *
+     * @param userId      unique identifier for requesting user, under which the request is performed.
+     * @param findRequest information object for find calls.
+     * @return list objects of the T type relevant in the findRequest information.
+     *
+     * @throws PropertyServerException    something went wrong with the REST call stack.
+     * @throws UserNotAuthorizedException the requesting user is not authorized to issue this request.
+     * @throws InvalidParameterException  one of the parameters is null or invalid.
+     */
    List<T> find(String userId, FindRequest findRequest) throws InvalidParameterException,
                                                                PropertyServerException,
                                                                UserNotAuthorizedException;
 
+    /**
+     * Update or replace a object.
+     *
+     * @param guid      unique identifier of the object to which the found objects should relate.
```

Comment:
I don't think this makes sense. I suggest just unique identifier of the object.

Reply:
No problem.